slsim/lens.py

```diff
@@ -517,69 +523,92 @@ def extended_source_magnitude(self, band, lensed=False):
+        return magnitude_list
 
     def point_source_magnification(self):
         """Macro-model magnification of point sources.
 
-        :return: signed magnification of point sources in same order as image positions
+        :return: list of signed magnification of point sources in same order as 
+         image positions.
         """
         if not hasattr(self, "_ps_magnification"):
             lens_model_list, kwargs_lens = self.deflector_mass_model_lenstronomy()
             lensModel = LensModel(lens_model_list=lens_model_list)
-            img_x, img_y = self.point_source_image_positions()
-            self._ps_magnification = lensModel.magnification(img_x, img_y, kwargs_lens)
-        return self._ps_magnification
+            self._ps_magnification_list = []
+            for image_pos in self.point_source_image_positions():
+                img_x, img_y = image_pos
+                self._ps_magnification_list.append(lensModel.magnification(img_x, img_y,
+                                                             kwargs_lens))
+        return self._ps_magnification_list 
 
     def extended_source_magnification(self):
         """Compute the extended lensed surface brightness and calculates the integrated
-        flux-weighted magnification factor of the extended host galaxy.
+        flux-weighted magnification factor of each extended host galaxy .
 
-        :return: integrated magnification factor of host magnitude
+        :return: list of integrated magnification factor of host magnitude
         """
+        #TODO: add source redshift in ray_shooting. Wait for lenstronomy new version.
         if not hasattr(self, "_extended_source_magnification"):
             kwargs_model, kwargs_params = self.lenstronomy_kwargs(band=None)
-            lightModel = LightModel(
-                light_model_list=kwargs_model.get("source_light_model_list", [])
-            )
-            lensModel = LensModel(
-                lens_model_list=kwargs_model.get("lens_model_list", [])
-            )
-            theta_E = self.einstein_radius
-            center_source = self.source.extended_source_position(
-                center_lens=self.deflector_position, draw_area=self.test_area
-            )
+            theta_E_list = self.einstein_radius
+            self._extended_source_magnification_list = []
+            # loop through each source.
+            for index, source in enumerate(self.source):
+                _light_model_list = kwargs_model.get(
+                        "source_light_model_list", [])[index]
+                kwargs_source_mag = [kwargs_params["kwargs_source"][index]]
+                if isinstance(_light_model_list, list):
+                    light_model_list = _light_model_list
+                else:
+                    light_model_list = [_light_model_list]
+                lightModel = LightModel(
+                    light_model_list=light_model_list)
+                lensModel = LensModel(
```

Comment:
the LensModel() instance needs to know the z_source_convention and z_source in it